picoquic/quicctx.c

```diff
@@ -1933,7 +1941,9 @@ int picoquic_renew_path_connection_id(picoquic_cnx_t* cnx, picoquic_path_t* path
     int ret = 0;
     picoquic_remote_cnxid_t * stashed = NULL;
 
-    if (cnx->remote_parameters.migration_disabled != 0 ||
+    if ((cnx->remote_parameters.migration_disabled != 0 &&
+        path_x->p_remote_cnxid != NULL &&
+        path_x->p_remote_cnxid->sequence >= cnx->retire_cnxid_before)||
         cnx->local_parameters.migration_disabled != 0) {
```

Comment:
Why does this need to check remote_parameters.migration_disabled or local_parameters.migration_disabled at all? In my mind, connection migration is distinct from switching connection IDs. Connection migration may require using a new connection ID, but issuing a new connection ID/retiring all connection IDs doesn't mean that a connection is migrating.

Reply:
The spec has evolved over time. Early versions more or less required to keep the same CID throughout the connection if the migration was disabled. Of course, there are now different controls for "voluntary migration" and for "number of CID accepted from the peer". I could remove these checks, but I will need first clarifications inside the WG. Maybe make that a different issue?

Reply:
Sounds good. I'll let you open that issue if thats ok.

Reply:
Created issue #1151